sdk/keyvault/azure-keyvault-keys/tests/test_key_client.py

```diff
@@ -22,10 +29,19 @@
 all_api_versions = get_decorator()
 only_hsm = get_decorator(only_hsm=True)
 only_hsm_7_3_preview = get_decorator(only_hsm=True, api_versions=[ApiVersion.V7_3_PREVIEW])
+only_vault_7_3_preview = get_decorator(only_vault=True, api_versions=[ApiVersion.V7_3_PREVIEW])
 logging_enabled = get_decorator(logging_enable=True)
 logging_disabled = get_decorator(logging_enable=False)
 
 
+def _assert_rotation_policies_equal(p1, p2):
+    assert p1.id == p2.id
+    assert p1.expires_in == p2.expires_in
+    assert p1.created_on == p2.created_on
+    assert p1.updated_on == p2.updated_on
+    assert len(p1.lifetime_actions) == len(p2.lifetime_actions)
```

Comment:
Totally up to you - but I would suggest testing that the lifetime_actions are deeply equal (or whatever is equivalent in python). Is that possible? What do you think? 

My reasoning is just concern over deserialization / transformation missing some data - either on input or output - which testing the length of the actions won't discover

Reply:
That's true, I would like to test for that. My reason for not including a thorough comparison was because I wasn't sure if we can assume some ordering of these lists, and testing for equality if they're unordered gets a bit messy for a helper method. The better solution is probably to just test deep equality on a case-by-case basis so I know what to expect and don't have to write messy, generalized logic. I'll do that, thanks for the suggestion!

Reply:
Improving these tests caught a bug with how I was handling generated lifetime action objects, so thank you for pointing this out! You saved me from future pain 🤠

Reply:
Awesome! 🎉